command/agent/command_test.go

```diff
@@ -142,3 +143,101 @@ func TestCommand_MetaConfigValidation(t *testing.T) {
 		}
 	}
 }
+
+func TestCommand_NullCharInDatacenter(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "nomad")
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	tcases := []string{
+		"char-\\000-in-the-middle",
+		"ends-with-\\000",
+		"\\000-at-the-beginning",
+	}
+	for _, tc := range tcases {
+		configFile := filepath.Join(tmpDir, "conf1.hcl")
+		err = ioutil.WriteFile(configFile, []byte(`
+        datacenter = "`+tc+`"
+        client{
+			enabled = true
+    	}`), 0600)
+		if err != nil {
+			t.Fatalf("err: %s", err)
+		}
+
+		// Make a new command. We preemptively close the shutdownCh
+		// so that the command exits immediately instead of blocking.
+		ui := new(cli.MockUi)
```

Comment:
According to https://github.com/hashicorp/nomad/issues/9026 this should be `cli.NewMockUi()`

Reply:
done, thanks 👍